api/api.py

```diff
@@ -9,99 +9,204 @@
+        listhandler.NotesListHandler, name='notes'
+    ),
 
+    webapp2.Route(
+        r'/api/collections/curators',
+        collectionshandler.CollectionsHandler, handler_method='curators', methods=['GET']
+    ),
+    webapp2.Route(
+        r'/api/<cont_name:collections>',
+        collectionshandler.CollectionsHandler, name='colls', handler_method='get_all', methods=['GET']
+    ),
+    webapp2.Route(
+        r'/api/<cont_name:collections>',
+        collectionshandler.CollectionsHandler, methods=['POST']
+    ),
+    webapp2.Route(
+        _format(r'/api/<cont_name:collections>/<cid:{cid_re}>'),
+        collectionshandler.CollectionsHandler, name='coll_details', methods=['GET', 'PUT', 'DELETE']
+    ),
+    webapp2.Route(
+        _format(r'/api/<cont_name:collections>/<cid:{cid_re}>/sessions'),
+        collectionshandler.CollectionsHandler, name='coll_ses', handler_method='get_sessions', methods=['GET']
+    ),
+    webapp2.Route(
+        _format(r'/api/<cont_name:collections>/<cid:{cid_re}>/acquisitions'),
+        collectionshandler.CollectionsHandler, name='coll_acq', handler_method='get_acquisitions', methods=['GET']
+    ),
 
-with open(os.path.join(os.path.dirname(__file__), 'schema.json')) as fp:
-    schema_dict = json.load(fp)
-for cls in [
-        users.Group,
-        users.User,
-        ]:
-    cls.post_schema = copy.deepcopy(schema_dict[cls.__name__.lower()])
-    cls.put_schema = copy.deepcopy(cls.post_schema)
-    cls.put_schema['properties'].pop('_id', None)
-    cls.put_schema.pop('required')
+    webapp2.Route(
+        _format(r'/api/users/<uid:{user_id_re}>/<cont_name:{cont_name_re}>'),
+        containerhandler.ContainerHandler, name='user_conts', handler_method='get_all_for_user', methods=['GET']
+    ),
+    webapp2.Route(
+        r'/api/projects/groups',
+        containerhandler.ContainerHandler, handler_method='get_groups_with_project', methods=['GET']
+    ),
+    webapp2.Route(
+        _format(r'/api/<par_cont_name:groups>/<par_id:{group_id_re}>/<cont_name:projects>'),
+        containerhandler.ContainerHandler, name='cont_sublist', handler_method='get_all', methods=['GET']
+    ),
+    webapp2.Route(
+        _format(r'/api/<par_cont_name:{cont_name_re}>/<par_id:{cid_re}>/<cont_name:{cont_name_re}>'),
+        containerhandler.ContainerHandler, name='cont_sublist', handler_method='get_all', methods=['GET']
+    ),
+    webapp2.Route(
+        _format(r'/api/<cont_name:{cont_name_re}>'),
+        containerhandler.ContainerHandler, name='cont_list', handler_method='get_all', methods=['GET']
+    ),
+    webapp2.Route(
+        _format(r'/api/<cont_name:{cont_name_re}>'),
+        containerhandler.ContainerHandler, methods=['POST']
+    ),
+    webapp2.Route(
+        _format(r'/api/<cont_name:{cont_name_re}>/<cid:{cid_re}>'),
+        containerhandler.ContainerHandler, name='cont_details', methods=['GET','PUT','DELETE']
```

Comment:
Do we use the "name=" param in the code for every route? If not, might be able to eliminate them to reduce visual noise.